Replace Donate page with more extensive Funding page

Calinou opened the pull request:
This is a preliminary version of the Godot Development Fund page I've been working on as part of my contract.

Other changes done as part of this PR:

- Replace the Patreon bar with a simpler highlighted **Donate** button.
- Move the list of sponsors to a reusable template to avoid duplication across pages.
- Add more utility classes and move some of them to `main.css`.
- Update the Patreon plugin to return the number of patrons and the unformatted amount of money in cents. The `number_format` Twig filter is more appropriate for number formatting.

The number of corporate sponsors and amount of money donated outside Patreon should be configured here: https://github.com/Calinou/godot-website/blob/75a5380cdf81ecdfae2be199c9fa7fe78bbc7f54/themes/godotengine/pages/fund.htm#L8-L21

This closes https://github.com/godotengine/godot-website/issues/102.

## Preview

*Click to view images at full size.*

### Home

<details>
<summary>Show</summary>

![](https://i.imgur.com/VIcCDva.png)
</details>

### About

<details>
<summary>Show</summary>

![](https://i.imgur.com/euKYSHk.jpg)
</details>

### Grants

*WIP: Needs more information on ongoing and past contracts to make this page more complete.*

![](https://i.imgur.com/3P23uFK.png)

### Corporate memberships

*WIP: Needs more text.*

![](https://i.imgur.com/vqlxkcK.png)

Calinou committed:
Replace Donate page with more extensive Funding page

Sslaxx commented on themes/godotengine/pages/fund/about.htm:
```diff
@@ -0,0 +1,126 @@
+<p>
+  It offers a complete editor, dedicated 2D and 3D engines, animation tools,
+  multi-platform exports, plugins, a Free asset store, and so much more! All
+  that in a lightweight (30 MB) package that runs smoothly on Windows, Mac and
```
30MB? Standard edition's download size maybe. I think that description needs to be clearer about what the size is referring to. Also note that the Mono version is larger due to its dependencies.

At the moment, that description makes it sound like the *whole thing* is 30MB installed. Which was never true.

Standard 3.2.3, with export templates, is about 670MB installed (with its binary weighing in at about 70MB of that on Linux). And while 670MB installed is over 20x the size of the 30MB download, it's still *considerably* smaller than either Unity or Unreal's installtion sizes.

You're always going to get trolls, but a more accurate description here is giving them less ammo (let's face it, they'll use *anything* as a reason to troll).

Calinou replied:
@Sslaxx I took the description as-is from the Patreon page. Also, considering you only need to download export templates once you're ready to publish your game (or "test drive" it at least), I don't think this statement is inaccurate or misleading. It's not like you *have* to install export templates as soon as you're creating your first script.

You could see it as a parallel to streaming videos: you don't need to download the entire video to begin watching a streamed video.

> You're always going to get trolls, but a more accurate description here is giving them less ammo (let's face it, they'll use anything as a reason to troll).

At some point, you have to stop catering to them :slightly_smiling_face:

Calinou commented:
Some updates:

- Rebased against the latest `master` branch.
- Added automatic dark theme support and made sure that new all pages look correct with both themes.
- Updated the Grants page with the latest hires.

coppolaemilio commented:
Considering the branch conflicts and that we are going to be refreshing this among other pages, should we close this PR @Calinou?
I think the work done can be a great start for the updated page but not sure if we should keep the PR open.
What do you think?

Calinou commented:
Closing per @coppolaemilio's comment.
